core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -889,24 +914,90 @@ private RealChannel(String authority) {
     }
 
     @Override
-    public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(MethodDescriptor<ReqT, RespT> method,
-        CallOptions callOptions) {
+    public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(
+        final MethodDescriptor<ReqT, RespT> method, final CallOptions callOptions) {
+      if (configSelector.get() != INITIAL_PENDING_SELECTOR) {
+        return newClientCall(method, callOptions);
+      }
+      final Context context = Context.current();
+      final PendingCall<ReqT, RespT> pendingCall = new PendingCall<>(context, method, callOptions);
+
+      syncContext.execute(new Runnable() {
+        @Override
+        public void run() {
+          exitIdleMode();
+          if (configSelector.get() == INITIAL_PENDING_SELECTOR) {
+            if (pendingCalls.isEmpty()) {
+              inUseStateAggregator.updateObjectInUse(RealChannel.this, true);
+            }
+            pendingCalls.add(pendingCall);
+          } else {
+            pendingCall.pendingCallRunnable.run();
+          }
+        }
+      });
+      return pendingCall;
+    }
+
+    @Override
+    public String authority() {
+      return authority;
+    }
+
+    private final class PendingCall<ReqT, RespT> extends DelayedClientCall<ReqT, RespT> {
+      final Runnable pendingCallRunnable;
+
+      PendingCall(
+          final Context context, final MethodDescriptor<ReqT, RespT> method,
+          final CallOptions callOptions) {
+        super(getCallExecutor(callOptions), scheduledExecutor, callOptions.getDeadline());
+        class PendingCallRunnable implements Runnable {
+          @Override
+          public void run() {
+            getCallExecutor(callOptions).execute(
+                new ContextRunnable(context) {
+                  @Override
+                  public void runInContext() {
+                    setCall(newClientCall(method, callOptions));
+                  }
+                }
+            );
+          }
+        }
+
+        pendingCallRunnable = new PendingCallRunnable();
+      }
+
+      @Override
+      void setCall(ClientCall<ReqT, RespT> call) {
```

Comment:
This is only called from the PendingCallRunnable. `cancel()` calls `setRealCall()`, not this method.

Reply:
Should I override `setRealCall()` or `cancel(final Status status, boolean onlyCancelPendingCall)`?

Reply:
Cancel would be ideal, but I don't think it currently makes sense because it might not actually cancel. setRealCall() is called under a lock, so that's no good. I think it may be good to have a new `protected void callCancelled() {}` noop method for overriding, that is called during `cancel(Status, boolean)` only if it cancels.

Reply:
Good idea. Done.